cob_cartesian_controller/include/cob_cartesian_controller/cartesian_controller_data_types.h

```diff
@@ -64,9 +68,140 @@ namespace cob_cartesian_controller
         unsigned int move_type;
         MoveLinStruct move_lin;
         MoveCircStruct move_circ;
+        ProfileStruct profile;
     };
 
-}//namespace
 
 
+    class PathArray{
+        public:
+            PathArray(double idx, double Se, double start_value, std::vector<double> array):
+                Se_(Se),
+                array_(array),
+                start_value_(start_value)
+            {
+                idx_= idx;
+                calcTe_=false;
+            }
+
+            ~PathArray()
+            {
+                array_.clear();
+            }
+
+            unsigned int getIdx();
+            bool getCalcTe();
+            double getSe();
+            std::vector<double> getArray();
+            void setCalcTe(bool calcTe);
+            void setIdx(unsigned int idx);
+            void setArray(std::vector<double> array);
+            double getStartValue();
+
+        private:
+            unsigned int idx_;
+            bool calcTe_;
+            double Se_;
+            std::vector<double> array_;
+            double start_value_;
+
+    };
+
+    inline double PathArray::getStartValue()
+    {
+        return start_value_;
+    }
+
+    inline void PathArray::setArray(std::vector<double> array)
+    {
+        array_ = array;
+    }
+
+    inline void PathArray::setCalcTe(bool calcTe)
+    {
+        calcTe_ = calcTe;
+    }
+
+    inline void PathArray::setIdx(unsigned int idx)
+    {
+        idx_=idx;
+    }
+
+    inline unsigned int PathArray::getIdx()
+    {
+        return idx_;
+    }
+
+    inline bool PathArray::getCalcTe()
+    {
+        return calcTe_;
+    }
+
+    inline double PathArray::getSe()
+    {
+        return Se_;
+    }
+
+    inline std::vector<double> PathArray::getArray()
+    {
+        return array_;
+    }
+
+
+    class PathMatrix{
+            public:
+                PathMatrix(PathArray &pa1,
+                           PathArray &pa2,
+                           PathArray &pa3,
+                           PathArray &pa4)
+//                           PathArray &pa5,
+//                           PathArray &pa6)
+                {
+                    pm_.push_back(pa1);
+                    pm_.push_back(pa2);
+                    pm_.push_back(pa3);
+                    pm_.push_back(pa4);
+//                    pm_.push_back(pa5);
+//                    pm_.push_back(pa6);
+                }
+                ~PathMatrix()
+                {
+                    pm_.clear();
+                }
+
+                std::vector<PathArray> getSortedMatrix();
+
+            private:
+                void sortMatrixRows();
+                std::vector<PathArray> pm_;
+
+        };
+
+    inline std::vector<PathArray> PathMatrix::getSortedMatrix()
+    {
+        sortMatrixRows();
+        return pm_;
+    }
+
+    inline void PathMatrix::sortMatrixRows()
```

Comment:
can we maybe use a better data container than vector with bubble-sort? maybe http://www.cplusplus.com/reference/list/list/sort/